config/src/inline/mod.rs

```diff
@@ -0,0 +1,141 @@
+mod conf_parser;
+pub use conf_parser::{ConfParser, ConfParserError};
+use ethers_solc::{artifacts::Node, ProjectCompileOutput};
+use serde_json::Value;
+use std::{
+    collections::{BTreeMap, HashMap},
+    path::Path,
+};
+
+/// Represents per-test configurations, declared inline
+/// as structured comments in Solidity test files. This allows
+/// to create configs directly bound to a solidity test.
+/// `T` is the configuration type, and is bound to the [`ConfParser`] trait. Known
+/// implementations of [`ConfParser`] include [`FuzzConfig`](super::FuzzConfig) and
+/// [`InvariantConfig`](super::InvariantConfig))
+#[derive(Default, Debug, Clone)]
+pub struct InlineConfig<T>
+where
+    T: ConfParser + 'static,
+{
+    /// Maps a (contract, test-function)
+    /// to a specific configuration provided by the user.
+    configs: HashMap<(String, String), T>,
```

Comment:
it can be confusing after a while what values these keys represent.

I'd like a helper type for this, then we can also implement `Borrow` so we can use `&(&str, &str)` as key in `get()` etc.

Reply:
> it can be confusing after a while what values these keys represent.

I can add a type for the key. This would improve readability:
```rust
/// Represents a (test-contract, test-function) pair
type InlineConfigKey = (String, String);
```

> I'd like a helper type for this, then we can also implement Borrow so we can use &(&str, &str) as key in get() etc.

Probably this is not required. The map is private and a meaningful (I hope ^^) accessor fn is provided . External users are not allowed to call `get()` on the map. 

https://github.com/foundry-rs/foundry/blob/7e3f10a5b471a9e6a95b62c13ddaaeb5eb4c0831/config/src/inline/mod.rs#L38-L40


Does it work for you? If not can you provide a more detailed example?